vpc/service/long_lived_lock.go

```diff
@@ -19,6 +24,81 @@ const (
 	timeBetweenTryingToAcquireLocks = 15 * time.Second
 )
 
+func (vpcService *vpcService) readLocks(rows *sql.Rows) ([]*vpcapi.Lock, error) {
+	locks := []*vpcapi.Lock{}
+	for rows.Next() {
+
+		lock := vpcapi.Lock{}
+		var heldUntil time.Time
+		err := rows.Scan(&lock.Id, &lock.LockName, &lock.HeldBy, &heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		lock.HeldUntil, err = ptypes.TimestampProto(heldUntil)
+		if err != nil {
+			return nil, err
+		}
+
+		locks = append(locks, &lock)
+	}
+
+	return locks, nil
+}
+
+func (vpcService *vpcService) GetLocks(ctx context.Context, empty *empty.Empty) (*vpcapi.GetLocksResponse, error) {
+	rows, err := vpcService.db.QueryContext(ctx, "SELECT id, lock_name, held_by, held_until FROM long_lived_locks LIMIT 1000")
+	if err != nil {
+		return nil, err
+	}
+	defer rows.Close()
+
+	locks, err := vpcService.readLocks(rows)
+	if err != nil {
+		return nil, err
+	}
+
+	return &vpcapi.GetLocksResponse{Locks: locks}, nil
+}
+
+func (vpcService *vpcService) GetLock(ctx context.Context, id *vpcapi.LockId) (*vpcapi.Lock, error) {
+	rows, err := vpcService.db.QueryContext(ctx, "SELECT id, lock_name, held_by, held_until FROM long_lived_locks WHERE id = $1", id.GetId())
```

Comment:
I think you can simplify this to QueryRowContext (because this could return a maximum of one row).

And then if err == sql.ErrNoRows, you can go directly to the lock not found error. 

This may result in a *slight* bit of code duplication, if you don't break out readLocks -> readLock

Reply:
You get this guarantee from the db because id is a PRIMARY / unique key:

```
titusvpcservice=> \d long_lived_locks
;                                        Table "public.long_lived_locks"
   Column   |            Type             | Collation | Nullable |                   Default                    
------------+-----------------------------+-----------+----------+----------------------------------------------
 id         | bigint                      |           | not null | nextval('long_lived_locks_id_seq'::regclass)
 lock_name  | text                        |           |          | 
 held_by    | text                        |           |          | 
 held_until | timestamp without time zone |           |          | 
Indexes:
    "long_lived_locks_pk" PRIMARY KEY, btree (id)
    "long_lived_locks_lock_name_uindex" UNIQUE, btree (lock_name)

```

Reply:
Yeah, I got lazy and wanted to reuse some logic but I see what you mean.